framework/src/base/MooseApp.C

```diff
@@ -279,6 +298,64 @@ MooseApp::MooseApp(InputParameters parameters)
 
   if (_check_input && isParamValid("recover"))
     mooseError("Cannot run --check-input with --recover. Recover files might not exist");
+
+  if (isParamValid("start_in_debugger"))
+  {
+    auto command = getParam<std::string>("start_in_debugger");
+
+    Moose::out << "Starting in debugger using: " << command << std::endl;
+
+    auto hostname = MooseUtils::hostname();
+
+    std::stringstream command_stream;
+
+    // This will start XTerm and print out some info first... then run the debugger
+    command_stream << "xterm -e \"echo 'Rank: " << processor_id() << "  Hostname: " << hostname
+                   << "  PID: " << getpid() << "'; echo ''; ";
+
+    // Figure out how to run the debugger
+    if (command.find("lldb") != std::string::npos || command.find("gdb") != std::string::npos)
+      command_stream << command << " -p " << getpid();
+    else
+      mooseError("Unknown debugger: ",
+                 command,
+                 "\nIf this is truly what you meant then contact moose-users to have a discussion "
+                 "about adding your debugger.");
+
+    // Finish up the command
+    command_stream << "\""
+                   << " & ";
+
+    std::string command_string = command_stream.str();
+    Moose::out << "Running: " << command_string << std::endl;
+
+    system(command_string.c_str());
+
+    // Sleep to allow time for the debugger to attach
+    std::this_thread::sleep_for(std::chrono::seconds(10));
+  }
+
+  if (!parameters.isParamSetByAddParam("stop_for_debugger"))
```

Comment:
Hmm, this doesn't seem quite right. You set 30 seconds as the default, but If I read this logic correctly, we won't go into this block if the user doesn't supply that parameter. Did you mean to **not** have a default when you declared this parameter?

Reply:
Nope - this code does the right thing.

What it does is you can support _just_ `--stop-for-debugger` (without providing a value) and it will stop for 30 seconds.  Or you can provide a time like `--stop-for-debugger=10`.

Reply:
Oh I see: `isParamSetByAddParam()` gets set to false if the command line option is supplied (whether a value is there or not). I suppose that's what it's supposed to do, I guess I'm surprised that it actually works properly for command line options - NICE!